[DONE] OOS-1120:  Hoist read-design-manifest.sh reuse branch above both-externals-down branch in /implement Step 1

## Out-of-Scope Observation

**Surfaced by**: Main agent (surfaced by Codex generic, round 3)
**Phase**: implement
**Vote tally**: N/A — accepted by main agent in quick mode

## Description

`skills/implement/SKILL.md` Step 1 normal mode — the both-externals-down inline-plan branch (currently at line ~552) executes BEFORE the reusable design manifest check (`read-design-manifest.sh` invocation, currently at line ~568). On a resumed session where `MANIFEST_OK=true` AND `SESSION_ID` matches `$IMPLEMENT_TMPDIR/session-id` AND both externals (`codex_available=false AND cursor_available=false`) happen to be unavailable, the orchestrator falls through to the both-externals-down inline-plan path — overwriting `plan.txt` / `voting-tally.md` with a degraded inline plan and silently discarding the prior `/design` artifact set (plan-review tally, contested criteria, rejected findings, architecture diagram). This is pre-existing behavior (independent of #1120's simplicity-classification gate, which now guards its own path via the preamble in Step 1 normal mode). Suggested fix: hoist the manifest-reuse `read-design-manifest.sh` invocation + reuse branch to be the FIRST sub-step in normal mode (immediately after `### Normal mode (quick_mode=false)` heading), so manifest reuse is evaluated BEFORE simplicity classification AND BEFORE the both-externals-down branch. Then the simplicity-classification preamble and the both-externals-down branch can stop carrying duplicate manifest-reuse skip language. Discovered by Codex generic reviewer in round 3 of the #1120 quick-mode review loop.

---
*This issue was automatically created by the larch `/implement` workflow from an out-of-scope observation surfaced during the workflow.*

**PR**: https://github.com/character-ai/larch/pull/1168

<!-- larch:implement-anchor v1 issue=1126 -->
<!-- section:plan-goals-test -->
# Implementation Plan — Hoist read-design-manifest.sh manifest-reuse check above both-externals-down branch

## Goal

In `skills/implement/SKILL.md` `### Normal mode (quick_mode=false)`, hoist the reusable-design-manifest check (currently embedded inside the standard `/design` path at lines 557-569) so it becomes the FIRST sub-step of normal mode — running BEFORE the simplicity-classification preamble AND BEFORE the both-externals-down inline-plan branch.

## Bug being fixed

On a resumed session where:
- `read-design-manifest.sh` would emit `MANIFEST_OK=true` AND `SESSION_ID` matches `$IMPLEMENT_TMPDIR/session-id` (manifest is reusable), AND
- `codex_available=false AND cursor_available=false AND design_only=false` (both externals unavailable)

the current ordering causes the both-externals-down branch (line 551) to fire FIRST. That branch overwrites `$IMPLEMENT_TMPDIR/design-export/plan.txt` and `voting-tally.md` with a degraded inline plan and silently discards the prior `/design` artifact set (plan-review tally, contested criteria, rejected findings, architecture diagram) the resume contract is written to preserve.

The simplicity-classification preamble (line 537) already encodes a manifest-reuse skip condition that duplicates the manifest-reuse logic; that duplicate language is the symptom of the missing hoist.

## Files to modify

- `skills/implement/SKILL.md` (single file)

## Edits in detail

### Edit 1 — Insert hoisted manifest-reuse block

After `### Normal mode (quick_mode=false)` heading and immediately after the existing `> **Continue after child returns.**` reminder paragraph (currently line 532), and BEFORE the `**Simplicity classification preamble — skip conditions**` block, insert the new sub-step. Phrase it as the FIRST sub-step of normal mode:

```
**Manifest reuse (resumed sessions — runs first)**: before any other normal-mode sub-step, check for a reusable design manifest. This guard runs BEFORE simplicity classification and BEFORE the both-externals-down inline-plan branch so a resumed session never overwrites the prior `/design` artifact set.

```bash
${CLAUDE_PLUGIN_ROOT}/skills/design/scripts/read-design-manifest.sh --implement-tmpdir "$IMPLEMENT_TMPDIR"
```

Parse stdout without `eval`/`source`. The reuse heuristic is a two-way conjunction:
1. `MANIFEST_OK=true`.
2. `PLAN_FILE` is non-empty and points to an existing non-empty file, AND `SESSION_ID` matches the value in `$IMPLEMENT_TMPDIR/session-id`.

(Session binding is enforced by `SESSION_ID` equality alone — `TIMESTAMP` is informational only and MUST NOT gate reuse, since the session-tmpdir lifetime already bounds the manifest's validity window.)

If both are true, reuse the manifest and proceed to Step 2 with **all manifest file variables** set from the reader output — not just `PLAN_FILE`, but also `PLAN_REVIEW_TALLY_FILE`, `CONTESTED_CRITERIA_FILE`, `OOS_FILE`, `REJECTED_FINDINGS_FILE`, `ACCEPTED_PLAN_FINDINGS_FILE`, and `ARCHITECTURE_DIAGRAM_FILE` (when present). Same surface as the post-`/design` success branch below; without this, downstream steps lose plan-review tally / rejected findings / architecture diagram on a resumed run.

Otherwise (no reusable manifest), continue with the normal-mode flow below (simplicity classification preamble, then the both-externals-down branch or the standard `/design` invocation).
```

### Edit 2 — Simplify the simplicity-classification preamble

In the existing `**Simplicity classification preamble — skip conditions**` block (currently at line 534), remove the second bullet (`- No reusable design manifest is present. If read-design-manifest.sh would emit ...`). The hoisted manifest-reuse block above already short-circuits to Step 2 on a reusable manifest, so this duplicate skip condition is no longer reachable.

Update the lead sentence from "classification runs only when both of these hold" to "classification runs only when `design_only=false`" — there is now only one condition.

Update the body sentence in the next paragraph: change "**Simplicity classification**: when both preamble conditions hold, classify the task..." to "**Simplicity classification**: when the preamble condition holds, classify the task...".

### Edit 3 — Strip the now-redundant manifest-reuse block from the standard /design path

In the paragraph beginning `Otherwise (at least one of \`codex_available\` / \`cursor_available\` is \`true\`, OR \`design_only=true\` and the bail above did not fire), proceed with the standard /design path:` (line 555), drop the trailing colon and merge with what follows so that:

- The intervening sentences `Before invoking /design, check for a reusable design manifest:` + the bash code block + `Parse stdout without eval/source. ...` + the two-step heuristic list + the `(Session binding is enforced...)` parenthetical + `If both are true, reuse the manifest and proceed to Step 2 with all manifest file variables set ... resumed run.` are REMOVED entirely.
- The `Otherwise invoke /design via the Skill tool. Canonical invocation order: ...` continuation (and the rest of the line-569 paragraph including `--auto`, `--subagent`, `--step-prefix`, `--branch-info`, `--session-env`, post-/design re-read of read-design-manifest.sh) is preserved.

Concretely, after the edit the paragraph reads:

```
Otherwise (at least one of `codex_available` / `cursor_available` is `true`, OR `design_only=true` and the bail above did not fire), invoke `/design` via the Skill tool. Canonical invocation order: `[--auto] [--subagent] --step-prefix "1.::design plan" --branch-info "IS_MAIN=$IS_MAIN IS_USER_BRANCH=$IS_USER_BRANCH USER_PREFIX=$USER_PREFIX CURRENT_BRANCH=$CURRENT_BRANCH" --session-env $IMPLEMENT_TMPDIR/session-env.sh <FEATURE_DESCRIPTION>`. Prepend `--auto` only if `auto_mode=true`. ... [rest unchanged]
```

## Behavior preservation

- **Resumed session, manifest reusable, both externals down**: NEW — manifest reuse fires first (Edit 1), short-circuits to Step 2 with all manifest file variables set. Bug fixed.
- **Resumed session, manifest reusable, externals up**: NEW — manifest reuse fires first (Edit 1), short-circuits to Step 2. Same effective result as before (the old standard-/design-path manifest-reuse branch did the same thing); now decided one block earlier.
- **Fresh session, no manifest, externals up**: hoisted block parses `MANIFEST_OK=false` (or `PLAN_FILE` empty), falls through. Simplicity classification runs (preamble's `design_only=false` condition still holds). Either auto-switches to quick mode (SIMPLE) or invokes /design via standard path. Unchanged.
- **Fresh session, no manifest, both externals down**: hoisted block falls through. Simplicity classification runs (may auto-switch). If still normal mode, both-externals-down branch fires and produces inline plan. Unchanged.
- **--design-only path**: hoisted block runs first; on a resumed `--design-only` session with reusable manifest, short-circuits to Step 2 (which then short-circuits to design-only's own publish path). On no manifest, simplicity classification preamble's `design_only=false` condition rejects, classification skipped. Both-externals-down + design_only=true bail still fires. Unchanged.

## Test plan

1. `make lint` — ensures markdown / shell linters pass.
2. `bash scripts/test-implement-rebase-macro.sh` — pins the rebase-checkpoint macro structure; should still pass since we don't touch macro call sites.
3. `bash scripts/test-fix-issue-bail-detection.sh` — pins Step 5a invariants in `/fix-issue`; unaffected.
4. Visual readback of the edited Step 1 normal-mode block: confirm the hoisted block appears immediately after the heading + Continue-after-child-returns reminder; the simplicity-classification preamble has only one bullet; the standard /design path no longer has a "Before invoking /design, check for a reusable design manifest" block.
[TRUNCATED — plan-goals-test exceeded 8000 chars]
<!-- section-end:plan-goals-test -->
<!-- section:plan-review-tally -->
Quick mode — no plan review voting.
<!-- section-end:plan-review-tally -->
<!-- section:code-review-tally -->
<!-- section-end:code-review-tally -->
<!-- section:diagrams -->
## Architecture Diagram

Architecture diagram not available.

## Code Flow Diagram

(Code Flow Diagram skipped — quick mode)
<!-- section-end:diagrams -->
<!-- section:version-bump-reasoning -->
# Version Bump Reasoning

- **Base commit**: `f97e4d9` (Pin post-merge anti-halt reminder literals (closes #1143) (#1164))
- **Current version**: `15.11.9`
- **Classification scope**: `skills/**` and `agents/**` only (public plugin surface).

## Result: PATCH

- **New version**: `15.11.10`

### PATCH rationale

No MAJOR or MINOR evidence found in the public plugin surface. Defaulting to PATCH per policy ("every PR must bump at least PATCH").

<!-- section-end:version-bump-reasoning -->
<!-- section:oos-issues -->
## Accepted Out-of-Scope Observations (filed)

- [#1165](https://github.com/character-ai/larch/issues/1165) — Add structural test pinning manifest-reuse before simplicity-classification + both-externals-down in /implement Step 1 (surfaced by Cursor-Testing)

## Rejected / Out-of-Scope Observations (not filed)

_None._
<!-- section-end:oos-issues -->
<!-- section:execution-issues -->
<details><summary>Execution Issues</summary>

### Pre-existing Code Issues
- **Step 5 (round 1)**: `scripts/anchor-section-markers.sh` `# shellcheck disable=SC2034` comment block lists only `tracking-issue-write.sh` and `assemble-anchor.sh` as `SECTION_MARKERS` consumers; the file header (and PR #1160) added `hydrate-anchor.sh` as a third consumer. Stale comment, no functional impact. One-line follow-up.

</details>
<!-- section-end:execution-issues -->
<!-- section:run-statistics -->
## Run Statistics

| Metric | Value |
| --- | --- |
| Mode | `--quick --merge` |
| Coder | codex |
| Plan-review rounds | 0 (quick mode) |
| Code-review rounds | 2 (quick mode loop converged) |
| OOS issues filed | 1 created, 0 deduplicated |
| Tracking issue | #1126 |
<!-- section-end:run-statistics -->

--- a/.claude-plugin/plugin.json
+++ b/.claude-plugin/plugin.json
@@ -1,6 +1,6 @@
 {
   "name": "larch",
-  "version": "15.11.9",
+  "version": "15.11.10",
   "description": "Multi-agent workflow automation for Claude Code: design, code review, implementation, and PR automation. Code review runs a 6-reviewer specialist panel (5 Cursor specialists + 1 Codex generic) — extended to 7 with an optional Gemini reviewer when the Gemini CLI is installed and healthy — with 3-voter adjudication; plan review runs a 3-reviewer panel (Claude + Codex + Cursor); design sketches run a 9-agent diverge-then-converge phase in regular mode (1 Claude + 4 Cursor + 4 Codex) or 3 in quick mode.",
   "author": {
     "name": "Sergey Zhupanov",

--- a/CHANGELOG.md
+++ b/CHANGELOG.md
@@ -5,6 +5,12 @@ All notable changes to this project will be documented in this file.
 The format is based on [Keep a Changelog](https://keepachangelog.com/en/1.1.0/),
 and this project adheres to [Semantic Versioning](https://semver.org/spec/v2.0.0.html).
 
+## [15.11.10] - 2026-05-05
+
+### Changed
+
+- Hoist the `read-design-manifest.sh` reusable-design-manifest check above the both-externals-down inline-plan branch in `/implement` Step 1 normal mode so a resumed session never overwrites the prior `/design` artifact set when both externals happen to be unavailable. Closes #1126.
+
 ## [15.11.9] - 2026-05-05
 
 ### Changed

--- a/skills/implement/SKILL.md
+++ b/skills/implement/SKILL.md
@@ -547,9 +547,7 @@ If both are true, reuse the manifest and proceed to Step 2 with **all manifest f
 
 Otherwise (no reusable manifest), continue with the normal-mode flow below (simplicity classification preamble, then the both-externals-down branch or the standard `/design` invocation).
 
-**Simplicity classification preamble — skip conditions**: classification runs only when `design_only=false`; otherwise skip it entirely and continue with the normal-mode flow below.
-
-- `design_only=false`. `--design-only` is mutually exclusive with quick mode and must not auto-switch into the degraded inline-plan path.
+**Simplicity classification preamble — skip condition**: classification runs only when `design_only=false`; otherwise skip it entirely and continue with the normal-mode flow below. (`--design-only` is mutually exclusive with quick mode and must not auto-switch into the degraded inline-plan path.)
 
 **Simplicity classification**: when the preamble condition holds, classify the task before invoking `/design`. Use `FEATURE_DESCRIPTION` plus a light codebase scan (Read / Grep / Glob of the obvious target files) to decide whether the work is SIMPLE.
 

--- a/skills/implement/SKILL.md
+++ b/skills/implement/SKILL.md
@@ -531,12 +531,27 @@ Proceed to Step 2.
 
 > **Continue after child returns.** When the child Skill returns, execute the NEXT step — do NOT end the turn, and do NOT write a summary, handoff, or "returning to parent" message. See `${CLAUDE_PLUGIN_ROOT}/skills/shared/subskill-invocation.md` section Anti-halt continuation reminder. (Branch-specific: applies only to the `/design` invocation in normal mode.)
 
-**Simplicity classification preamble — skip conditions**: classification runs only when both of these hold; otherwise skip it entirely and continue with the normal-mode flow below.
+**Manifest reuse (resumed sessions — runs first)**: before any other normal-mode sub-step, check for a reusable design manifest. This guard runs BEFORE simplicity classification and BEFORE the both-externals-down inline-plan branch so a resumed session never overwrites the prior `/design` artifact set.
+
+```bash
+${CLAUDE_PLUGIN_ROOT}/skills/design/scripts/read-design-manifest.sh --implement-tmpdir "$IMPLEMENT_TMPDIR"
+```
+
+Parse stdout without `eval`/`source`. The reuse heuristic is a two-way conjunction:
+1. `MANIFEST_OK=true`.
+2. `PLAN_FILE` is non-empty and points to an existing non-empty file, AND `SESSION_ID` matches the value in `$IMPLEMENT_TMPDIR/session-id`.
+
+(Session binding is enforced by `SESSION_ID` equality alone — `TIMESTAMP` is informational only and MUST NOT gate reuse, since the session-tmpdir lifetime already bounds the manifest's validity window.)
+
+If both are true, reuse the manifest and proceed to Step 2 with **all manifest file variables** set from the reader output — not just `PLAN_FILE`, but also `PLAN_REVIEW_TALLY_FILE`, `CONTESTED_CRITERIA_FILE`, `OOS_FILE`, `REJECTED_FINDINGS_FILE`, `ACCEPTED_PLAN_FINDINGS_FILE`, and `ARCHITECTURE_DIAGRAM_FILE` (when present). Same surface as the post-`/design` success branch below; without this, downstream steps lose plan-review tally / rejected findings / architecture diagram on a resumed run.
+
+Otherwise (no reusable manifest), continue with the normal-mode flow below (simplicity classification preamble, then the both-externals-down branch or the standard `/design` invocation).
+
+**Simplicity classification preamble — skip conditions**: classification runs only when `design_only=false`; otherwise skip it entirely and continue with the normal-mode flow below.
 
 - `design_only=false`. `--design-only` is mutually exclusive with quick mode and must not auto-switch into the degraded inline-plan path.
-- No reusable design manifest is present. If `read-design-manifest.sh` would emit `MANIFEST_OK=true` with `PLAN_FILE` non-empty + readable AND `SESSION_ID` matching `$IMPLEMENT_TMPDIR/session-id` (the same conjunction the manifest reuse branch below evaluates), skip simplicity classification — auto-switching to quick mode on a resumed run would discard the prior `/design` artifact set (plan-review tally, contested criteria, rejected findings, architecture diagram) the resume contract is written to preserve.
 
-**Simplicity classification**: when both preamble conditions hold, classify the task before invoking `/design`. Use `FEATURE_DESCRIPTION` plus a light codebase scan (Read / Grep / Glob of the obvious target files) to decide whether the work is SIMPLE.
+**Simplicity classification**: when the preamble condition holds, classify the task before invoking `/design`. Use `FEATURE_DESCRIPTION` plus a light codebase scan (Read / Grep / Glob of the obvious target files) to decide whether the work is SIMPLE.
 
 A task qualifies as SIMPLE only when all of these are true:
 - Small surface area: expected edits are localized to one or a few files with obvious ownership.
@@ -552,21 +567,7 @@ When the task is not SIMPLE: leave `quick_mode=false` and continue with the norm
 
 The `design_only=false` gate is load-bearing: `--design-only`'s contract is to publish design artifacts (plan, plan-review tally, diagrams, OOS) to the tracking issue as the run's deliverable. It is mutually exclusive with `--quick` precisely because quick mode produces a degraded plan with no plan-review voting. Inheriting that degradation here when externals are down would silently violate the same contract. When `codex_available=false AND cursor_available=false AND design_only=true`, do NOT skip /design — print `**⚠ 1: design plan — both Codex and Cursor unavailable but --design-only requires external-backed plan-review. Bailing to cleanup.**`, set `STALL_TRACKING=true`, and skip to Step 18.
 
-Otherwise (at least one of `codex_available` / `cursor_available` is `true`, OR `design_only=true` and the bail above did not fire), proceed with the standard /design path:
-
-Before invoking `/design`, check for a reusable design manifest:
-
-```bash
-${CLAUDE_PLUGIN_ROOT}/skills/design/scripts/read-design-manifest.sh --implement-tmpdir "$IMPLEMENT_TMPDIR"
-```
-
-Parse stdout without `eval`/`source`. The skip-/design heuristic is a two-way conjunction:
-1. `MANIFEST_OK=true`.
-2. `PLAN_FILE` is non-empty and points to an existing non-empty file, AND `SESSION_ID` matches the value in `$IMPLEMENT_TMPDIR/session-id`.
-
-(Session binding is enforced by `SESSION_ID` equality alone — `TIMESTAMP` is informational only and MUST NOT gate reuse, since the session-tmpdir lifetime already bounds the manifest's validity window.)
-
-If both are true, reuse the manifest and proceed to Step 2 with **all manifest file variables** set from the reader output — not just `PLAN_FILE`, but also `PLAN_REVIEW_TALLY_FILE`, `CONTESTED_CRITERIA_FILE`, `OOS_FILE`, `REJECTED_FINDINGS_FILE`, `ACCEPTED_PLAN_FINDINGS_FILE`, and `ARCHITECTURE_DIAGRAM_FILE` (when present). Same surface as the post-`/design` success branch below; without this, downstream steps lose plan-review tally / rejected findings / architecture diagram on a resumed run. Otherwise invoke `/design` via the Skill tool. Canonical invocation order: `[--auto] [--subagent] --step-prefix "1.::design plan" --branch-info "IS_MAIN=$IS_MAIN IS_USER_BRANCH=$IS_USER_BRANCH USER_PREFIX=$USER_PREFIX CURRENT_BRANCH=$CURRENT_BRANCH" --session-env $IMPLEMENT_TMPDIR/session-env.sh <FEATURE_DESCRIPTION>`. Prepend `--auto` only if `auto_mode=true`. Append `--subagent` (after `--auto`, before `--step-prefix` in argv order) only if `inline_mode=false` (default); when `inline_mode=true`, omit `--subagent` so /design's heavy phase runs in /design's in-turn context (execution topology only — parent verbosity suppression unchanged). After `/design` returns, immediately run `read-design-manifest.sh --implement-tmpdir "$IMPLEMENT_TMPDIR" --emit-load-breadcrumb` again; if it does not emit `MANIFEST_OK=true`, print `**⚠ 1: design plan — design manifest unavailable: $ERROR. Bailing to cleanup.**`, set `STALL_TRACKING=true`, and skip to Step 18. On success, the reader appends `📥 1: design plan — manifest loaded (plan=<basename>)` as the trailing line of its stdout (after the `KEY=value` envelope) — that breadcrumb is the orchestrator's first mid-Step-1 visible line; set `PLAN_FILE` and all manifest file variables from the reader output.
+Otherwise (at least one of `codex_available` / `cursor_available` is `true`, OR `design_only=true` and the bail above did not fire), invoke `/design` via the Skill tool. Canonical invocation order: `[--auto] [--subagent] --step-prefix "1.::design plan" --branch-info "IS_MAIN=$IS_MAIN IS_USER_BRANCH=$IS_USER_BRANCH USER_PREFIX=$USER_PREFIX CURRENT_BRANCH=$CURRENT_BRANCH" --session-env $IMPLEMENT_TMPDIR/session-env.sh <FEATURE_DESCRIPTION>`. Prepend `--auto` only if `auto_mode=true`. Append `--subagent` (after `--auto`, before `--step-prefix` in argv order) only if `inline_mode=false` (default); when `inline_mode=true`, omit `--subagent` so /design's heavy phase runs in /design's in-turn context (execution topology only — parent verbosity suppression unchanged). After `/design` returns, immediately run `read-design-manifest.sh --implement-tmpdir "$IMPLEMENT_TMPDIR" --emit-load-breadcrumb` again; if it does not emit `MANIFEST_OK=true`, print `**⚠ 1: design plan — design manifest unavailable: $ERROR. Bailing to cleanup.**`, set `STALL_TRACKING=true`, and skip to Step 18. On success, the reader appends `📥 1: design plan — manifest loaded (plan=<basename>)` as the trailing line of its stdout (after the `KEY=value` envelope) — that breadcrumb is the orchestrator's first mid-Step-1 visible line; set `PLAN_FILE` and all manifest file variables from the reader output.
 
 > **Continue after child returns.** When `/design` returns, execute the Cross-Skill Health Update + `BRANCH_NAME` capture + Step 1.r rebase checkpoint + Step 2 breadcrumb in order — do NOT write a summary, handoff, or "returning to parent" message first. See `${CLAUDE_PLUGIN_ROOT}/skills/shared/subskill-invocation.md` section Anti-halt continuation reminder.
 

